api/services/control/control_bench_test.go

```diff
@@ -17,7 +17,7 @@ func BenchmarkMarshalVertex(b *testing.B) {
 	v := sampleVertex()
 	for i := 0; i < b.N; i++ {
 		var err error
-		Buf, err = proto.Marshal(v)
+		Buf, err = v.MarshalVT()
```

Comment:
Rather than replace these with a specific implementation, it might be good to have a centralized `Marshal` function that can be switched out to use different implementations.

Reply:
A potential issue with this is it could mask which implementation of marshal is being used. The type switch on the message type to determine if it supports `MarshalVT` would also incur an extra memory allocation due to the creation of an interface type.

We have some areas in the code where we explicitly use the original protobuf implementation because we need the deterministic marshaling of the original library.